src/main/java/com/adobe/acrobatsign/service/RestApiAgreements.java

```diff
@@ -849,4 +855,195 @@ public JSONObject workflowInfo(String accessToken, String workflowId) throws IOE
 		return responseJson;
 	}
 
+	public JSONObject getUsers(String accessToken) throws IOException{
+		// TODO Auto-generated method stub
+		
+		final String endpointUrl = getBaseURL() + GET_USERS;
+		final Map<String, String> headers = new HashMap<>();
+		headers.put(RestApiUtils.HttpHeaderField.AUTHORIZATION.toString(), accessToken);
+		
+		final JSONObject userDetails = (JSONObject) RestApiUtils.makeApiCall(endpointUrl,
+				RestApiUtils.HttpRequestMethod.GET, headers);
+		return userDetails;
+	}
+
+	public JSONObject getUserTemplate(String accessToken, String userEmail) throws IOException{
+		// TODO Auto-generated method stub
+		final String endpointUrl = getBaseURL() + GET_LIBRARY_TEMPLATES;
+		final Map<String, String> headers = new HashMap<>();
+		headers.put(RestApiUtils.HttpHeaderField.AUTHORIZATION.toString(), accessToken);
+		headers.put(RestApiUtils.HttpHeaderField.USER_EMAIL.toString(), "email:" + userEmail);
+		
+		final JSONObject userTemplates = (JSONObject) RestApiUtils.makeApiCall(endpointUrl,
+				RestApiUtils.HttpRequestMethod.GET, headers);
+		return userTemplates;
+	}
+
+	public String downloadTemplates(String accessToken, List<LibraryDocument> seletedTemplateList, String userEmail,
+			HttpServletResponse response) {
+		// TODO Auto-generated method stub
+		// URL to invoke the agreements end point.
+				final RestTemplate restTemplate = new RestTemplate();
+				String templateName = null;
+				// Create header list.
+				ZipOutputStream zos = null;
+				try {
+					final String endpointUrl = getBaseURL() + GET_LIBRARY_TEMPLATES;
+					zos = new ZipOutputStream(response.getOutputStream());
+					for (final LibraryDocument libraryDocument : seletedTemplateList) {
+						templateName = libraryDocument.getId() + "----" + libraryDocument.getName();
+						final StringBuilder urlString = new StringBuilder();
+						urlString.append(endpointUrl).append("/").append(libraryDocument.getId()).append("/combinedDocument");
+
+						final StringBuilder directoryPath = new StringBuilder();
+						directoryPath.append(downloadPath).append(libraryDocument.getOwnerEmail());
+
+						final File directory = new File(directoryPath.toString());
+						if (!directory.exists()) {
+							directory.mkdir();
+						}
+						// ByteArrayResource resource = new ByteArrayResource(Files.readAllBytes(path));
+						final HttpHeaders restHeader = new HttpHeaders();
+						restHeader.add(RestApiUtils.HttpHeaderField.AUTHORIZATION.toString(), accessToken);
+						restHeader.add(RestApiUtils.HttpHeaderField.CONTENT_TYPE.toString(), "application/json");
```

Comment:
there is a duplicate code. please create a common method to use the common code.